examples/chrono/chrono.cpp

```diff
@@ -0,0 +1,219 @@
+﻿/*
+ * Copyright (c) 2022, Christopher Kormanyos
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <chrono>
+
+#if defined(__GNUC__) && defined(__AVR__)
+#define ATTRIBUTE(...) __attribute__((__VA_ARGS__))
+#else
+#define ATTRIBUTE(X)
+#endif
+
+namespace mcal {
+
+namespace reg {
+
+// Bit values.
+constexpr auto bval5      = static_cast<std::uint8_t>(UINT8_C(1) << 5U);
+
+constexpr auto sfr_offset = static_cast<std::uint8_t>(UINT8_C(0x20));
+
+// System registers.
+constexpr auto mcusr      = static_cast<std::uint8_t>(UINT8_C(0x14) + sfr_offset);
+
+// Port registers.
+constexpr auto pinb       = static_cast<std::uint8_t>(UINT8_C(0x03) + sfr_offset);
+constexpr auto ddrb       = static_cast<std::uint8_t>(UINT8_C(0x04) + sfr_offset);
+constexpr auto portb      = static_cast<std::uint8_t>(UINT8_C(0x05) + sfr_offset);
+
+// Timer registers
+constexpr auto tifr0      = static_cast<std::uint8_t>(UINT8_C(0x15) + sfr_offset);
+constexpr auto timsk0     = static_cast<std::uint8_t>(UINT8_C(0x6E));
+constexpr auto tccr0b     = static_cast<std::uint8_t>(UINT8_C(0x25) + sfr_offset);
+constexpr auto tcnt0      = static_cast<std::uint8_t>(UINT8_C(0x26) + sfr_offset);
+
+// Watchdog registers
+constexpr auto wdtcsr     = static_cast<std::uint8_t>(UINT8_C(0x60));
+
+} // namespace reg
+
+namespace gpt {
+
+using value_type = std::uint64_t;
+
+auto init() -> void;
+
+auto get_time_elapsed() -> value_type;
+
+} // namespace gpt
+
+namespace wdg {
+
+auto init() -> void;
+
+inline void trigger()
+{
+  #if defined(__GNUC__) && defined(__AVR__)
+  asm volatile("wdr");
+  #endif
+}
+
+} // namespace gpt
+} // namespace mcal
+
+namespace local {
+
+mcal::gpt::value_type mcal_gpt_system_tick;
+
+using clock_type = std::chrono::steady_clock;
+
+auto delay_1s() noexcept -> void;
+
+} // namespace local
+
+auto main(void) -> int
```

Comment:
Why is `void` used here in the function parameter list?
As far as I know `int main(void)` and `int main()` would be completely equivalent in C++ (contrary to C).

Reply:
> Why is `void` used here in the function parameter list?

Good catch. The form of `main` in the example-under-review is I believe, in fact, wrong.

I will correct it to be one of... (see below).

In C++, the allowed forms are either

```C
int main();
```

```C
int main(int, char*);
```

Reply:
oops... Typo in the second form, I meant to write

```C
int main(int, char**);
```